docs/homepage.md

```diff
@@ -0,0 +1,35 @@
+# Welcome 👋
+
+Welcome to the AutoMapper documentation, it's a library that solve a simple problem: removing all the code you need to
+map one object to another. A boring code to write and often replaced by non-performant alternatives like the Symfony's
+Serializer.
+
+For that goal it provides an unique interface:
+```php
+interface AutoMapperInterface
+{
+    /**
+     * Maps data from a source to a target.
+     *
+     * @param array|object                      $source  Any data object, which may be an object or an array
+     * @param 'array'|class-string|array|object $target  To which type of data, or data, the source should be mapped
+     * @param array                             $context Mapper context
+     *
+     * @return array|object|null The mapped object
+     */
+    public function map(array|object $source, string|array|object $target, array $context = []): null|array|object;
+}
+```
+The source is from where the data comes from, it can be either an array or an object. The target is where the data 
+should be mapped to, it can be either a string (representing a type: array or class name) or directly an array or 
+object (in that case construction of the object is avoided).
```

Comment:
```suggestion
`$source` is the origin of the data, it can either be an array or an object. 
`$target` is where the data should be mapped to. It can be:
	- a string representing a type: either a class name or the string "array". In this case a new object (or array) will be created,
	- an instance of an object or an array: in this case construction of the object is avoided and your variable will be hydrated from the `$source`
```